usaspending_api/references/v2/views/recipients.py

```diff
@@ -68,22 +76,25 @@ def _prepare_search_terms(self, request_data: Dict[str, Union[str, List[str]]])
         Returns:
             A list containing the sanitized search text and recipient levels.
         """
-        fields = [request_data["search_text"], request_data.get("recipient_levels", [])]
+        fields = [request_data["search_text"], request_data.get("recipient_levels", []), request_data.get("duns")]
         return [es_sanitize(field).upper() if isinstance(field, str) else field for field in fields]
 
-    def _create_es_search(self, search_text: str, recipient_levels: List[str], limit: int) -> RecipientSearch:
+    def _create_es_search(
+        self, search_text: str, recipient_levels: List[str], duns: str, limit: int
+    ) -> RecipientSearch:
         """
         Create an Elasticsearch search query for recipient autocomplete.
 
         Args:
             search_text: The search text entered by the user.
             recipient_levels: The list of recipient levels to filter by.
+            duns: Any specific duns key specified by the user.
```

Comment:
I wasn't sure if that was needed to be repeated here since recipient_levels is also optional but the comment above doesn't exactly outline that. I suppose I can update the description for both.

Reply:
The str | None optional parameters syntax is only supported in python 3.10 and above. So unfortunately do not have the capacity to do that right now. I have left the type as None and left a comment in the code explaining that it is by default null unless a str value is passed in. I believe that is sufficient.

Reply:
Hmm no i think I'll just leave it as type string and remove the comment. Reading the api contract a dev should be able to clearly see what that the duns parameter is optional